src/sentry/workflow_engine/handlers/action/notification/notification.py

```
@@ -41,25 +41,38 @@ def execute(
     ) -> None:
         try:
             handler = group_type_notification_registry.get(detector.type)
-            handler(job, action, detector)
+            handler.handle_workflow_action(job, action, detector)
         except NoRegistrationExistsError:
             logger.exception(
                 "No notification handler found for detector type: %s",
                 detector.type,
                 extra={"detector_id": detector.id, "action_id": action.id},
             )
-            # Maybe metrics here?
 
 
 @group_type_notification_registry.register(ErrorGroupType.slug)
 class IssueAlertRegistryInvoker(LegacyRegistryInvoker):
-    def handle_workflow_action(self, job: WorkflowJob, action: Action, detector: Detector) -> None:
-        # TODO(iamrajjoshi): Implement this
-        pass
+    @staticmethod
+    def handle_workflow_action(job: WorkflowJob, action: Action, detector: Detector) -> None:
+        try:
+            handler = issue_alert_handler_registry.get(action.type)
+            handler.invoke_legacy_registry(job, action, detector)
+        except NoRegistrationExistsError:
+            logger.exception(
+                "No issue alert handler found for action type: %s",
+                action.type,
+                extra={"action_id": action.id},
+            )
```

Should this swallow the exception? Does the caller assume these calls never fail?

---

good point, let me raise exceptions, we can swallow if needed later